python/pyspark/worker.py

```diff
@@ -69,6 +69,7 @@ def chain(f, g):
 
 
 def wrap_udf(f, return_type):
+    f = fail_on_stopiteration(f)
```

Comment:
Should we move the wrapping to `read_single_udf` instead of putting in individual `wrap_*` functions?

Reply:
`wrap_grouped_map_pandas_udf` needs to call `argspec`, so we would need to check `evalType` twice in `read_single_udf`. But it's possible, yea

Reply:
Or just move the call to `get_argspec` in `read_single_udf`

Reply:
Can we wrap `fail_on_stopiteration` after `wrap*_udf`?

Reply:
>move the call to get_argspec in read_single_udf

Yeah I think that works